crates/goose-server/src/routes/config_management.rs

```diff
@@ -320,6 +323,73 @@ pub async fn providers(
     Ok(Json(providers_response))
 }
 
+#[utoipa::path(
+    post,
+    path = "/config/init",
+    responses(
+        (status = 200, description = "Config initialization check completed", body = String),
+        (status = 500, description = "Internal server error")
+    )
+)]
+pub async fn init_config(
+    State(state): State<AppState>,
+    headers: HeaderMap,
+) -> Result<Json<String>, StatusCode> {
+    verify_secret_key(&headers, &state)?;
+
+    let config = Config::global();
+    
+    // 200 if config already exists
+    if config.exists() {
+        return Ok(Json("Config already exists".to_string()));
+    }
+
+    // Find the workspace root (where the top-level Cargo.toml with [workspace] is)
+    let workspace_root = match std::env::current_exe() {
```

Comment:
@salman1993 @lily-de @baxen Let me know if there is a cleaner/simpler way to do this in rust. Will happily replace it in a followup PR!